src/libsyntax/parse/mod.rs

```diff
@@ -174,14 +174,25 @@ pub fn parse_stream_from_source_str(name: FileName, source: String, sess: &Parse
     source_file_to_stream(sess, sess.source_map().new_source_file(name, source), override_span)
 }
 
-// Create a new parser from a source string
+/// Create a new parser from a source string
```

Comment:
Shouldn't this use `maybe_new_parser_from_source_str` and emit the buffered errors? That would also allow removing `source_file_to_parser`, I think.

Reply:
How deep do you want me to go? Should i reimplement each existing function in terms of its new `maybe_` variant?

Reply:
If you have the time, that'd be great. I just want to minimize code duplication. Whenever I add a fallible version of an existing method, I rewrite the callees if under a dozen, or I rewrite the infallible version to call the fallible version.

Reply:
Makes sense. I've got the change written locally, i'll push once i finish my current build.